pkg/settings/setting.go

```diff
@@ -148,6 +151,7 @@ type Provider interface {
 	Set(name, value string) error
 	SetIfUnset(name, value string) error
 	SetAll(settings map[string]Setting) error
+	Delete(name string) error
```

Comment:
I would like to point that `Set` and `SetAll` functions are also destructive. All settings will be recreated when removed after restarting Rancher in opposite to situation if they would be modified by `Set`. They will stay in broken state. 

Please explain how users can use it from outside Rancher?

Reply:
From outside of rancher they can't update it. Users would need to have permission to delete, but we try to avoid anything where they can accidentally mess up the cluster. If they delete it and don't restart Rancher, then Rancher will break. Also if there were custom settings those would be lost. They could overwrite using update but that is also a legitimate thing they need to do.

Reply:
I have added one more condition to `Delete`, which is preventing from removing settings based on ENV_VARIABLE. Regarding removing settings, I have noticed that users can `Edit` and `Delete` settings from Rancher API. I am not sure if any operation or check are needed to prevent users to break cluster settings. The best would be to do not delete anything if we are worry that something can be broken. As I have checked users can `Update` and `Delete` any setting through Rancher API and this part doesn't change anything with it.